error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/JUnitClassModifiersTest.java

```diff
@@ -0,0 +1,133 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class JUnitClassModifiersTest {
+  private final CompilationTestHelper compilationTestHelper =
+      CompilationTestHelper.newInstance(JUnitClassModifiers.class, getClass());
+  private final BugCheckerRefactoringTestHelper refactoringTestHelper =
+      BugCheckerRefactoringTestHelper.newInstance(JUnitClassModifiers.class, getClass());
+
+  @Test
+  void identification() {
+    compilationTestHelper
+        .addSourceLines(
+            "A.java",
+            "import org.junit.jupiter.api.Test;",
+            "",
+            "// BUG: Diagnostic contains:",
+            "class A {",
+            "  @Test",
+            "  void foo() {}",
+            "}")
+        .addSourceLines(
+            "B.java",
+            "import org.junit.jupiter.params.ParameterizedTest;",
+            "",
+            "// BUG: Diagnostic contains:",
+            "class B {",
+            "  @ParameterizedTest",
+            "  void foo() {}",
+            "}")
+        .addSourceLines(
+            "C.java",
+            "import org.junit.jupiter.api.Test;",
+            "",
+            "// BUG: Diagnostic contains:",
+            "public class C {",
```

Comment:
This one is flagged because it's non-final; to test the `public` trigger the class should be `final`.